flyteplugins/go/tasks/pluginmachinery/ioutils/remote_file_output_reader.go

```diff
@@ -3,77 +3,232 @@ package ioutils
 import (
 	"context"
 	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
 
 	"github.com/pkg/errors"
 
 	"github.com/flyteorg/flyte/flyteidl/gen/pb-go/flyteidl/core"
 	"github.com/flyteorg/flyte/flyteplugins/go/tasks/pluginmachinery/io"
+	"github.com/flyteorg/flyte/flyteplugins/go/tasks/pluginmachinery/k8s"
 	"github.com/flyteorg/flyte/flytestdlib/storage"
 )
 
-type RemoteFileOutputReader struct {
-	outPath        io.OutputFilePaths
+type ErrorRetriever interface {
```

Comment:
`RemoveFileOutputReader` already implements the `OutputReader` interface. Why introduce a new one?

Reply:
ErrorRetriever is an internal interface used within the `RemoteFileOutputReader`. Remote file reader reads outputs + errors. The error retriever deals with the error reading aspect, based on the strategy.

Reply:
```
type RemoteFileOutputReader struct {
	outPath        io.OutputFilePaths
	store          storage.ComposedProtobufStore
	maxPayloadSize int64
	errorRetriever ErrorRetriever
}
```

Reply:
Make it into a private interface `errorRetriever`